svc/GridExportService.js

```diff
@@ -182,8 +182,10 @@ export class GridExportService {
     getCellData(gridModel, record, column) {
         const {field, exportValue} = column;
 
+        // If ag-grid calculated aggregates, it stores them in aggData.
+        const aggData = gridModel.agApi.getRowNode(record.id).aggData;
```

Comment:
Can we throw this logic into an `else` block below? I don't think we need to do this unless there isn't an `exportValue` configured

Reply:
Currently, we're looking up aggData for every cell in the grid.

To help minimize the number of calls to the agApi, we could get the `aggData` once in `getRecordRow()` above and pass it into `getCellData()`.

We could also check if `treeMode && record.children.length` to avoid getting it for leaf nodes / flat grids.

Reply:
I've put in those two optimizations.

Reply:
Regarding Tom's optimization about TreeMode -- wouldn't that prevent this from working when we are just using agGrid and grouping with non-full-width group rows?

Reply:
Can we define an aggregation function on something that isn't a TreeGrid?

Reply:
My bad here -- Haynes straightened me out -- yes we can create a purely "grouping" based aggregation, but our export is not going to see it because our export relies on records -> Different problem!